glib/Marshaller.cs

```diff
@@ -88,19 +88,19 @@ static string Utf8PtrToStringFast (IntPtr ptr, int len)
 			}
 		}
 
-		static bool hasFastGetStringOverload = typeof (System.Text.Encoding).GetMethod ("GetString", new [] { typeof (byte*), typeof (int) }) != null;
 		public static string Utf8PtrToString (IntPtr ptr) 
 		{
 			if (ptr == IntPtr.Zero)
 				return null;
 
 			int len = (int) (uint)glibsharp_strlen (ptr);
-			if (hasFastGetStringOverload)
-				return Utf8PtrToStringFast (ptr, len);
-
+#if HAVE_NET_4_6
```

Comment:
How about wrapping the old 3 lines with #if UTF8_SLOWPATH , and after them, `return Utf8PtrToStringFast (ptr, len);` this way removing this build check in the future will be easier and cause less diff noise